ingest/ingest_pipeline.py

```diff
@@ -310,12 +311,38 @@ def upload_metadata_to_bq(self):
                 return 1
         return 0
 
+    def load_expression_file(self, models, is_gene_model=False):
+        collection_name = self.matrix.COLLECTION_NAME
+        # Creates operations to bulk write function
+        bulk_operations = list(map(lambda model: InsertOne(model), models))
+        try:
+            if is_gene_model:
+                # bulk_write_results describes the type and count of operations performed.
+                bulk_write_results = self.db[collection_name].bulk_write(
+                    bulk_operations
+                )
+                # Succesfully wrote documents
+                return True, bulk_write_results
+            else:
+                bulk_write_results = self.db['data_arrays'].bulk_write(bulk_operations)
+                # Succesfully wrote documents
+                return True, bulk_write_results
+        except BulkWriteError as bwe:
+            self.error_logger.error(bwe.details, extra=self.extra_log_params)
+            return False, bwe.details
+
+        except Exception as e:
+            self.error_logger.error(e, extra=self.extra_log_params)
+            return False, None
+
     @trace
     @my_debug_logger()
     # @profile
     def ingest_expression(self) -> int:
         """Ingests expression files.
         """
+        gene_documents = []
```

Comment:
It seems like this might be one place where we are experiencing a memory leak - I don't see where the contents of these lists get emptied.  It looks as though these two lists would just continue to grow as we read more data.

Reply:
@eweitz has already identified that this is a place for memory concerns in his [doc](https://docs.google.com/document/d/1NobO9jDfX5NoAGxXoNdoqHoTZ83rTEJxL0IU_X91l-k/edit). This ticket addresses speed and not memory. Further work will need to be done to address the memory leaks as identified in [SCP-1916](https://broadworkbench.atlassian.net/browse/SCP-1916)

Reply:
OK, sounds good.